### PR TITLE
tasks.py: import wikimedia datas

### PR DESCRIPTION
**Pull request by remi-dupre**
Import of wikimedia dumps generated here: https://github.com/QwantResearch/wikimedia-dumps. This data is used to update the prioritization of the display of POI's.

Comparison with production: (after application of this branch and of https://github.com/QwantResearch/qwant-basic-gl-style/pull/87)

Z = 14, before | after
Capture d’écran_2019-11-18_13-29-50

Z = 15, before | after
Capture d’écran_2019-11-18_13-29-37

Z = 16, before | after
Capture d’écran_2019-11-18_13-29-24

### CODE REVIEW

File: import_data/invoke.yaml [hunk 1]
```diff
@@ -26,6 +26,21 @@ osm:
   ## Path of the file
   file:
 
+wikidata:
+  sitelinks:
+    url: https://github.com/QwantResearch/wikimedia-dumps/releases/download/10/2019/sitelinks.csv.gz
```
Comment (GuillaumeGomez) [inline import_data/invoke.yaml]: You download all wikimedia stuff from october 2019? :o

Comment (remi-dupre) [reply import_data/invoke.yaml]: I'll give more explicit names to these dumps, this one actually includes data from November to mid October.

File: .travis.yml [hunk 2]
```diff
@@ -1,23 +0,0 @@
...
-  - if [ "$TRAVIS_PULL_REQUEST_BRANCH" == "" ] && [ "$TRAVIS_BRANCH" == "master" ]; then
-    echo "$DOCKER_TOKEN" | docker login -u "$DOCKER_USERNAME" --password-stdin;
-    fi
-
-script:
-  - export DOCKER_IMAGE="${DOCKER_IMAGE_PREFIX}_$SERVICE"
-  - docker build --label "org.label-schema.vcs-ref=$TRAVIS_COMMIT" -t $DOCKER_IMAGE -f ./$SERVICE/Dockerfile .
-  - if [ "$TRAVIS_PULL_REQUEST_BRANCH" == "" ] && [ "$TRAVIS_BRANCH" == "master" ]; then
-    docker push $DOCKER_IMAGE;
-    fi
```
Comment (amatissart) [inline .travis.yml]: any reason to delete this file ?

File: import_data/tasks.py [hunk 3]
```diff
@@ -15,18 +19,35 @@
 from pydantic import BaseModel
 from pydantic.datetime_parse import parse_datetime
 
-
+psycopg2.extensions.register_adapter(dict, psycopg2.extras.Json)
```
Comment (amatissart) [inline import_data/tasks.py]: Out of curiosity, is it necessary to use both `register_adapter` and `register_hstore` here ?

File: import_data/tasks.py [hunk 4]
```diff
@@ -15,18 +19,35 @@
...
+        user=ctx.pg.user,
+        database=ctx.pg.database,
+        host=ctx.pg.host,
+        password=ctx.pg.password,
+        port=ctx.pg.port
+    )
+    psycopg2.extras.register_hstore(connection, globally=True)
+    return connection
+
+def _execute_sql(ctx, sql, db=None, additional_options="", istream=None):
```
Comment (amatissart) [inline import_data/tasks.py]: this new parameter `istream` is never used, is it ?

File: import_data/invoke.yaml [hunk 5]
```diff
@@ -26,6 +26,21 @@ osm:
...
+
+  labels:
+    url: https://github.com/QwantResearch/wikimedia-dumps/releases/download/10/2019/labels.csv.gz
+    file: labels.csv.gz
+    table: wd_names
+
+wikimedia_stats:
+  url: https://github.com/QwantResearch/wikimedia-dumps/releases/download/10/2019/stats.csv.gz
+  file: stats.csv.gz
+  table: wm_stats
```
Comment (amatissart) [inline import_data/invoke.yaml]: I think we will also need an option to enable/disable these imports, at least for debug purposes.

File: import_data/import-wikidata/wikidata_tables.sql [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                WHEN name = '' THEN -1000000
+                ELSE -10 * poi_class_rank(poi_class(subclass, mapping_key))
+            END
+            + (
+                SELECT COUNT(*)::integer
+                FROM each(tags)
+                WHERE key LIKE 'name:%'
+            )
+        )
+    )
```
Comment (amatissart) [inline import_data/import-wikidata/wikidata_tables.sql]: A few questions about this function:
 * should these values be hardcoded here ?
 * how to handle different scale of pageviews (eg on different time windows) ? etc.

File: import_data/tasks.py (hunk 3 above)
Comment (remi-dupre) [reply import_data/tasks.py]: Not a single one, this one is actually only intended for Postgres' JSON fields.

File: import_data/invoke.yaml (hunk 5 above)
Comment (remi-dupre) [reply import_data/invoke.yaml]: I put it as disabled by default, maybe I shoulded ?

The import isn't that long, it takes about 1 hour to import labels (the operation on hstores takes a bit more time) and the two other tasks only need a few minutes to run.

File: import_data/import-wikidata/wikidata_tables.sql [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+                              AND wm_stats.lang = wd_sitelinks.lang)
+            WHERE wd_sitelinks.id = tags->'wikidata';
+        RETURN CASE
+            WHEN views_count > min_views THEN
+                0.5 * (1 + LOG(LEAST(max_views, views_count)) / LOG(max_views))
+            WHEN name = '' THEN
+                0.0
+            ELSE
+                0.5 * (
+                    1 - poi_class_rank(poi_class(subclass, mapping_key))::real / 2000
```
Comment (amatissart) [inline import_data/import-wikidata/wikidata_tables.sql]: Not sure to understand where "2000" comes from.  
It seems to be related to rank values in some way ?
If the maximum value for class rank is 1000, does it mean the "poi_weight" for any poi with a name is >= 0.25 ?

We should try to be as explicit as possible about these subtle dependencies, for maintainability. 
Ideally there would be a single constant used in both contexts. If that is not possible, we can still define constant names and/or add comments about this dependency.

File: import_data/tasks.py [hunk 8]
```diff
@@ -185,6 +202,7 @@ def run_sql_script(ctx):
     # load several psql functions
     _run_sql_script(ctx, "import-sql/language.sql")
     _run_sql_script(ctx, "postgis-vt-util/postgis-vt-util.sql")
+    _run_sql_script(ctx, "import-wikidata/wikidata.sql")
```
Comment (amatissart) [inline import_data/tasks.py]: Is it `wikidata_tables.sql` ?

File: import_data/tasks.py [hunk 9]
```diff
@@ -315,7 +420,15 @@ def run_post_sql_scripts(ctx):
     logging.info("running postsql scripts")
     _run_sql_script(ctx, "generated_base.sql")
     _run_sql_script(ctx, "generated_poi.sql")
+    override_openmaptile_functions(ctx)
 
+@task
+def override_openmaptile_functions(ctx):
```
Comment (amatissart) [inline import_data/tasks.py]: typo
```suggestion
def override_openmaptiles_functions(ctx):
```

**Pull request merged**